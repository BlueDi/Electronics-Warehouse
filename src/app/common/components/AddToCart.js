import React, { Component } from 'react';
import { Button } from 'semantic-ui-react';
import { withCookies } from 'react-cookie';

class AddToCart extends Component {
  handleClick = () => {
    var newItems = this.props.items;
    delete newItems[0]['image'];
    var oldCart = this.props.cookies.get('cart') || [];
    var newUniqueItems = [];
    for (let item of newItems)
      if (!oldCart.some(i => i.id == item.id))
        newUniqueItems.push({ item, ammount: 1 });

    var newCart = [...oldCart, ...newUniqueItems];
    this.props.cookies.set('cart', newCart);
  };

  render() {
    var canRequest = this.props.cookies.get('can_request') === 'true';
    return (
      canRequest && <Button onClick={this.handleClick}>Add to Cart</Button>
    );
  }
}

export default withCookies(AddToCart);
