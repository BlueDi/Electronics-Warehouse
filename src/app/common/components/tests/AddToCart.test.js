import React from 'react';
import renderer from 'react-test-renderer';
import AddToCart from '../AddToCart';

var item1 = {
  id: 1,
  amount: 2,
  description: 'BMW motor',
  details: { details: 'Super detailed description', manufacturer: 'Multicomp' },
  image:
    'iVBORw0KGgoAAAANSUhEUgAAAG8AAABYCAYAAAD7uql1AAAgAElEQVR4Xu29e5RV5Znu+5vfN6/r\nVmtVFXWjoChABFERVCKgUUAjthFzMZqYxLjT3aP79D456d49zj5nZJzTo0f/0+fsHumT7O6xY/be\n2umklcSEaFRuIiiIF0AUFQVFQSAlRRVVRdW6zut3/phrLs…',
  last_edit: '2018-11-20T11:46:01.628Z',
  last_price: 89.12,
  location: 'HHD12U',
  reference: '213748227'
};
var item2 = {
  id: 2,
  amount: 3,
  description: 'Blue LED',
  details: { details: 'Super detailed description', manufacturer: 'Farnell' },
  image:
    'iVBORw0KGgoAAAANSUhEUgAAAG8AAABYCAYAAAD7uql1AAAgAElEQVR4Xu29e5RV5Znu+5vfN6/r\nVmtVFXWjoChABFERVCKgUUAjthFzMZqYxLjT3aP79D456d49zj5nZJzTo0f/0+fsHumT7O6xY/be\n2umklcSEaFRuIiiIF0AUFQVFQSAlRRVVRdW6zut3/phrLs…',
  last_edit: '2018-11-19T17:29:35.629Z',
  last_price: 1.98,
  location: 'DDS98K',
  reference: '68760203'
};

test('AddToCart Button Render', () => {
  const component = renderer.create(<AddToCart items={[item1, item2]} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
