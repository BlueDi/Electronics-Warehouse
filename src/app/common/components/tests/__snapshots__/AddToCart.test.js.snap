// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCart Button Render 1`] = `null`;
